### PR TITLE
Support installing live branches from the manifest

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This adds support to install and activate WooCommerce plugin versions acquired from the live branches manifest.

I have intentionally kept the UI quite simple still to avoid a lot of complexity and focus on the main features of allowing the install and activation of any plugin build from the live branches manifest.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

0. Add the `register_page` code back here: https://github.com/woocommerce/woocommerce/blob/d8b0a26b6e4d151030457f0761e918a6bd1fcdac/plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches.php#L23 (since we don't have feature flags)
1. Run this branch alongside an activated WooCommerce core plugin
2. Go to the menu WooCommerce > Live Branches
3. A list of live branches you can install should be listed. If they are not already installed they should display a button "Install and Activate"
4. Click "Install and Activate" on a plugin of your choice. A spinner should appear while it is downloading and activating the plugin.
5. Eventually this install should finish and the item you clicked should now show a greyed out button "Activated"
6. To swap to a different live branch you can click "Install and Activate" on another list item.
7. This should successfully install and swap to the live branch you just clicked.
8. Now if you do a page refresh (I think keeping track of the install state of all plugins could be a future enhancement? I wanted to keep scope small as possible) you should see one of your live branches is installed but not activated because it has a button that just says "Activate"
9. Click "Activate" and it should just perform activation for that plugin and deactivate the old one.

Screenshot 2022-12-19 at 6 56 35 PM

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-beta-tester/api/live-branches/install.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	)
+);
+
+register_woocommerce_admin_test_helper_rest_route(
+	'/live-branches/activate/v1',
+	'activate_version',
+	array(
+		'methods'             => 'POST',
+		'permission_callback' => function( $request ) {
+			// Avoid using WC functions as core will be deactivated during this request.
```
Comment (samueljseay) [inline plugins/woocommerce-beta-tester/api/live-branches/install.php:32]: When we go to activate a plugin version, there may not be an active version of WooCommerce installed. By overriding the permission_callback here and using core WP permission checks we ensure that the site does not crash during activation.

File: plugins/woocommerce-beta-tester/api/live-branches/manifest.php [hunk 2]
```diff
@@ -19,6 +21,13 @@
 function fetch_live_branches_manifest() {
 	$response = wp_remote_get( 'https://betadownload.jetpack.me/woocommerce-branches.json' );
 	$body     = wp_remote_retrieve_body( $response );
+	$installer = new WC_Beta_Tester_Live_Branches_Installer();
+
+	$obj = json_decode( $body );
+
+	foreach ( $obj->pr as $key => $value ) {
+		$value->install_status = $installer->check_install_status( $value->version );
```
Comment (samueljseay) [inline plugins/woocommerce-beta-tester/api/live-branches/manifest.php:29]: add plugin installation status into the manifest

File: plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches-installer.php [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+defined( 'ABSPATH' ) || exit;
+
+require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+const LIVE_BRANCH_PLUGIN_PREFIX = 'wc_beta_tester_live_branch';
+
+/**
+ * WC_Beta_Tester Live Branches Installer Class.
+ */
+class WC_Beta_Tester_Live_Branches_Installer {
```
Comment (samueljseay) [inline plugins/woocommerce-beta-tester/includes/class-wc-beta-tester-live-branches-installer.php:17]: This class handles install/activation/unzip etc of live branches.

File: plugins/woocommerce-beta-tester/src/live-branches/hooks/live-branches.tsx [hunk 4]
```diff
@@ -39,3 +42,101 @@ export const useLiveBranchesData = () => {
 
 	return { branches, isLoading: loading };
 };
+
+export const useLiveBranchInstall = (
```
Comment (samueljseay) [inline plugins/woocommerce-beta-tester/src/live-branches/hooks/live-branches.tsx:46]: This hook is basic and handles requests to manage the installation/activation of plugins

**Review by psealock — DISMISSED** [review]
Nice @samueljseay, this is all testing well as far as I can see.

>I think keeping track of the install state of all plugins could be a future enhancement?

This would be why you can have several "Active" buttons at once, I presume.

Code looks good and reads well

**Comment by samueljseay** [comment]
> This would be why you can have several "Active" buttons at once, I presume.

Yes I think I should do this as a follow up enhancement, but I wanted to make some decisions about how the UI should function before doing more about this.

@jonathansadowski and I discussed maybe making the UI similar to Jetpack Beta. Here's a screen shot from their UI:

image

**Comment by samueljseay** [comment]
@psealock apologies I missed that lint wasn't running in my environment. Have fixed lint issues and need a re-review 🙇🏻

**Review by psealock — APPROVED** [review]

**Pull request merged**